src/coreclr/nativeaot/Runtime/DebugHeader.cpp

```diff
@@ -0,0 +1,294 @@
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig1);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig2);
+
+    MAKE_SIZE_ENTRY(StressMsg);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, fmtOffsCArgs);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, facility);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, timeStamp);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, args);
+
+    MAKE_SIZE_ENTRY(Object);
+    MAKE_DEBUG_FIELD_ENTRY(Object, m_pEEType);
+
+    MAKE_SIZE_ENTRY(Array);
+    MAKE_DEBUG_FIELD_ENTRY(Array, m_Length);
+
+    MAKE_SIZE_ENTRY(RuntimeInstance);
+    MAKE_DEBUG_FIELD_ENTRY(RuntimeInstance, m_pThreadStore);
+
+    RuntimeInstance *g_pTheRuntimeInstance = GetRuntimeInstance();
+    MAKE_GLOBAL_ENTRY(g_pTheRuntimeInstance);
+
+    MAKE_GLOBAL_ENTRY(g_gcDacGlobals);
+
+    MAKE_GLOBAL_ENTRY(g_pFreeObjectEEType);
+
+    void *g_stressLog = &StressLog::theLog;
+    MAKE_GLOBAL_ENTRY(g_stressLog);
+
+    // Some DAC functions need to know the module base address, easiest way is with
+    // the HANDLE to our module which is the base address.
+    HANDLE moduleBaseAddress = PalGetModuleHandleFromPointer((void *)&PopulateDebugHeaders);
+    MAKE_GLOBAL_ENTRY(moduleBaseAddress);
+
+    static_assert(EEType::Flags::EETypeKindMask         == 0x0003, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::RelatedTypeViaIATFlag  == 0x0004, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::IsDynamicTypeFlag      == 0x0008, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::HasFinalizerFlag       == 0x0010, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::HasPointersFlag        == 0x0020, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::GenericVarianceFlag    == 0x0080, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::OptionalFieldsFlag     == 0x0100, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::IsGenericFlag          == 0x0400, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
+    static_assert(EEType::Flags::ElementTypeMask        == 0xf800, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
```

Comment:
Do we really have a dependency on `IsDynamicType` and `OptionalFieldsFlag`?

Reply:
No, I was just on auto pilot here